clang/test/SemaSYCL/undefined-functor.cpp

```diff
@@ -0,0 +1,50 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2020 -verify -fsyntax-only %s
+// This test checks that an error is thrown when a functor without a call operator defined is used as a kernel.
+
+#include "sycl.hpp"
+
+using namespace sycl;
+queue q;
+
+struct FunctorWithoutCallOperator; // expected-note {{forward declaration of 'FunctorWithoutCallOperator'}}
+
+struct StructDefined {
+  int x;
+};
+
+class FunctorWithCallOpDeclared {
```

Comment:
Nit: the names should be the other way round.   This one is 'FunctorWithCallOpDefined'.

Reply:
Duh! Good catch.